lib/crypto/tasks.py

```diff
@@ -24,43 +24,51 @@ def __exit__(self, type, value, traceback):
 
 
 @task
-def sign_addons(addon_ids, **kw):
+def sign_addons(addon_ids, force=False, **kw):
     log.info('[{0}] Signing addons.'.format(addon_ids))
     for version in Version.objects.filter(
             addon_id__in=addon_ids, addon__type=amo.ADDON_EXTENSION):
-        log.info('Signing addon {0}, version {1}'.format(version.addon,
-                                                         version))
-        # We need to bump the version number of the file and the version, so
+        # We need to bump the version number of the file and the Version, so
         # the Firefox extension update mecanism picks this new signed version
         # and installs it.
-        bump_version_number(version)
+        if force:
+            to_sign = version.files.all()
+        else:
+            to_sign = [f for f in version.files.all() if not f.is_signed]
+        if not to_sign:
+            log.info('Not signing addon {0}, version {1} (no files or already '
+                     'signed)'.format(version.addon, version))
+            continue
+        log.info('Signing addon {0}, version {1}'.format(version.addon,
+                                                         version))
         try:
-            version.sign_files()
-        except SigningError as e:
+            for file_obj in to_sign:
+                bump_version_number(file_obj)
+                sign_file(file_obj)
+            # Now update the Version model.
+            version.update(version='{0}.1'.format(version.version))
```

Comment:
Creating a new version is a bit more difficult, indeed, and poses some other issues, but most of those issues were relevant at the time we were signing the same addon possibly multiple times (what would we do with the original files? Do we move them somewhere else? Signing multiple times would mean overwriting the previous original file with the file we signed just before...).

Maybe we want to reassess now... at the end of the day, we're still going to irk devs by modifying their files (or adding new ones).

@clouserw, thoughts?